src/system/storage.rs

```diff
@@ -138,7 +138,7 @@ where
 
     /// Retrieves the data from the storage based on hashes of the data [`segments`][Segment],
     /// or Error(NotFound) if some of the hashes were not present in the base.
-    pub fn retrieve(&self, request: &[Hash]) -> io::Result<Vec<Vec<u8>>> {
+    pub fn retrieve(&self, request: Vec<&Hash>) -> io::Result<Vec<Vec<u8>>> {
```

Comment:
```suggestion
    pub fn retrieve(&self, request: &[&Hash]) -> io::Result<Vec<Vec<u8>>> {
```
и снизу можно будет убрать .iter().collect() который не к месту

Reply:
снизу keys: &Vec<K> из TargetChunk, там не поменять